hw_4_5.js

```diff
@@ -0,0 +1,362 @@
+// function trimString() {
+//     let d = a.slice(+(b - 1 ),+(c));
+//     return d;
+//   }
+// alert(trimString());
+
+// //Task_4
+// let a = prompt("Введите число");
+// let b = 0;
+// function getSumNumbers() {
+//     while (a != 0) {
+//        c = a % 10;
+//        b = b + c;
+//        a = (a - c) / 10;
+//     }
+//     return b;
+// }
+// alert(getSumNumbers());
+
+//Task_5
+// let a = prompt("Введите первое число ...");
+// let b = prompt("Введите второе число ...");
+// let e = 0;
+//     function getSum(){
+//         if (a === b){
+//             return(a);
+//         }
+//         else if (a < b){
+//             while(a <= b){
+//                 e = e + +a;
+//                 a++;
+//             }
+//             return(e);
+//         }
+//         else if (a > b){
+//             while(b <= a){
+//                 e = e + +b;
+//                 b++;
+//             }
+//             return(e);
+//     }
+// }
+// if(Number.isInteger(+a) && Number.isInteger(+b) ){
+//     alert(getSum());
+// }
+// else    {
+//     alert("Числа не целые");
+// }
+
+// //Task_6
+// let a = prompt('Введите True или False');
+// a = a.toLowerCase();
+// function Foo() {
+// 	console.log('Foo')
+// }
+// function Boo() {
+// 	console.log('Boo')
+// }
+// function FooBoo() {
+// 	if (a == 'true') {
+// 		return (Foo());
+// 	}
+// 	else if (a == 'false') {
+// 		return (Boo);
+// 	}
+// }
+// console.log(FooBoo());
+
+
+// Task 7
+
+// const user = {
+// 	name: "Tanya",
+// 	gender: "woman"
+// };
+
+// console.log(`name: ${user.name} \ngender: ${user.gender}`);
+// delete user.name
+// delete user.gender
+// console.log(`name: ${user.name} \ngender: ${user.gender}`);
+
+
+
+// Task 8
+// const user = {
+// 	name: "Tanya",
+// 	gender: "woman"
+// };
+
+// console.log("name" in user);
+
+
+// Task 9
+
+// const student = {
+// 	name: 'John',
+// 	age: 19,
+// 	isHappy: true
+// }
+
+
+// function keys(Obj) {
+// 	for (let key in Obj) {
+// 		console.log(key);
+// 	}
+// }
+
+// function values(Obj) {
+// 	for (let key in Obj) {
+// 		console.log(Obj[key]);
+// 	}
+// }
+
+
+// keys(student);
+// values(student);
+
+
+
+// Task 10
+
+
+// const colors = {
+// 	'ru pum pu ru rum': {
+// 		red: 'красный',
+// 		green: 'зеленый',
+// 		blue: 'синий'
+// 	};
+// };
+
+
+// console.log(colors['ru pum pu ru rum'].red);
+// console.log(colors['ru pum pu ru rum'].blue);
+
+
+// Task 11
+
+// let salaries = {
+// 	andrey: 500,
+// 	sveta: 413,
+// 	anton: 987,
+// 	tanya: 664,
+// 	alexandra: 199
+// };
+
+
+// function countSalary(Obj) {
+// 	let a = 0;
+// 	let b = 0;
+// 	for (let i in Obj) {
+// 		a++;
+// 		b = b + Obj[i];
+// 	}
+// 	return b / a;
+// }
+
+
+// let averageSalary = countSalary(salaries);
+// console.log(averageSalary)
+
+
+
+// // Task 12
+
+// function valid(login, password) {
+// 	this.login = login.trim();
+// 	this.login = login.toLowerCase();
+// 	this.password = password.trim();
+// 	this.password = password.toLowerCase();
```

Comment:
зачем два раз присваивать, тем более trim перезатрется?
можно так сделать
```suggestion
// 	this.login = login.trim().toLowerCase();
// 	this.password = password.trim().toLowerCase();
```